File: tests/co-simulations/static/0/c-sync/tb.c
#include <assert.h>
#include <stdio.h>

#include "cosim_iface.h"

#include "afbd.h"
#include "main.h"
#define AFBD_IFACE &iface


int main(int argc, char *argv[]) {
	assert(argc == 3);

	afbd_iface_t iface = cosim_iface_iface();

	cosim_iface_init(argv[1], argv[2], NULL);

	uint32_t id;
	afbd_read(main_ID, &id);
	if (id != afbd_main_ID) {
		fprintf(stderr, "read wrong ID %x, expecting %x\n", id, afbd_main_ID);
		cosim_iface_end(1);
	}

	uint32_t timestamp;
	afbd_read(main_TIMESTAMP, &timestamp);
	if (timestamp != afbd_main_TIMESTAMP) {
		fprintf(stderr, "read wrong TIMESTAMP %x, expecting %x\n", id, afbd_main_TIMESTAMP);
		cosim_iface_end(1);
	}

	cosim_iface_end(0);

	return 0;
}

File: tests/co-simulations/static/0/python/tb.py
import sys
import traceback

import cosim
import afbd


WRITE_FIFO_PATH = sys.argv[1]
READ_FIFO_PATH = sys.argv[2]
REG_JSON = sys.argv[3]

CLK_PERIOD = 10

iface = cosim.Iface(WRITE_FIFO_PATH, READ_FIFO_PATH)

try:
    main, _ = afbd.generate(iface, REG_JSON)

    id = main.ID.read()
    assert id == main.ID.value, f"Read wrong ID {id}, expecting {main.ID.value}"
    print(f"ID: {id}\n")

    ts = main.TIMESTAMP.read()
    assert ts == main.TIMESTAMP.value, f"Read wrong TIMESTAMP {ts}, expecting {main.TIMESTAMP.value}"
    print(f"Timestamp: {ts}\n")

    iface.end(0)

except Exception as E:
    iface.end(1)
    print(traceback.format_exc())
